src/plugins/robots/foot-bot/simulator/footbot_light_rotzonly_sensor.cpp

```diff
@@ -166,7 +166,7 @@ namespace argos {
                    * Increase magnitude of result of division to ensure correct rounding
                    */
                   Real fIdx = (cAngleLightWrtFootbot - SENSOR_HALF_SPACING) / SENSOR_SPACING;
-                  SInt32 nReadingIdx = (fIdx > 0) ? fIdx + 0.5f : fIdx - 0.5f;
+                  SInt32 nReadingIdx = static_cast<UInt32>((fIdx > 0) ? fIdx + 0.5f : fIdx - 0.5f);
```

Comment:
Are these casts really necessary? Is it a warning that makes them necessary? Also, this cast is wrong because it loses the sign.

Reply:
Same thing here--will fix the cast and push